src/coreclr/tests/src/Loader/classloader/MethodImpl/CovariantReturns/Validation/OverrideSameSigAsDecl.il

```diff
@@ -0,0 +1,483 @@
+    ldarg.1
+    ldstr "NonGenThroughGen4 TestType.NonGenThroughGenFunc()"
+    stind.ref
+    ldnull
+    ret
+  }
+
+  .method public hidebysig newslot virtual instance class GenToNonGen3<!V,object> GenToNonGenOverride(string& res)
+  {
+    .override method instance class B class GenBaseType<!U,!V>::GenToNonGen(string& res)
+    ldarg.1
+    ldstr "GenToNonGen3<!V,object> TestType.GenToNonGenOverride()"
+    stind.ref
+    ldnull
+    ret
+  }
+
+  .method public hidebysig newslot virtual instance class NonGenericDerived4 NewGenFunc1(string& res)
+  {
+    .override method instance class GenRetType<int32,object> class GenBaseType<!U,!V>::MyGenFunc(string& res)
+    ldarg.1
+    ldstr "NonGenericDerived4 TestType.NewGenFunc1()"
+    stind.ref
+    ldnull
+    ret
+  }
+
+  .method public hidebysig newslot virtual instance class GenDerive3<!U,!V> NewGenFunc2(string& res)
+  {
+    .override method instance class GenRetType<!0, class Dictionary<string,!1>> class GenBaseType<!U,!V>::MyGenFunc(string& res)
+    ldarg.1
+    ldstr "GenDerive3<!U,!V> TestType.NewGenFunc2()"
+    stind.ref
+    ldnull
+    ret
+  }
+  
+  .method public hidebysig newslot virtual instance class GenRetType<!U,!V> NewFunc1(string& res)
+  {
+    .override method instance object class GenBaseType<!U,!V>::MyFunc(string& res)
+    ldarg.1
+    ldstr "GenRetType<!U,!V> TestType.NewFunc1()"
+    stind.ref
+    ldnull
+    ret
+  }
+  
+  .method public hidebysig newslot virtual instance class C NewFunc2(string& res)
+  {
+    .override method instance class B class GenBaseType<!U,!V>::MyFunc(string& res)
+    ldarg.1
+    ldstr "C TestType.NewFunc2()"
+    stind.ref
+    ldnull
+    ret
+  }
+
+  .method public hidebysig specialname rtspecialname instance void  .ctor() cil managed
+  {
+    .maxstack  8
+    ldarg.0
+    call       instance void class GenBaseType<!U,!V>::.ctor()
+    ret
+  }
+}
+
+// ========================================================================================
+
+.class public auto ansi beforefieldinit Invalid1<U,V> extends class GenTestType<!U,!V>
+{
+  .custom instance void [System.Private.CoreLib]System.Runtime.CompilerServices.ValidateMethodImplRemainsInEffectAttribute::.ctor() = (01 00 00 00)
+    
+  .method public hidebysig newslot virtual instance class B NonGenThroughGenOverride_Invalid(string& res)
+  {
+    .override method instance class B class GenBaseType<!U,!V>::NonGenThroughGenFunc(string& res)
+    ldnull
+    ret
+  }
+
+  .method public hidebysig specialname rtspecialname instance void  .ctor() cil managed
+  {
+    .maxstack  8
+    ldarg.0
+    call       instance void class GenTestType<!U,!V>::.ctor()
+    ret
+  }
+}
+
+.class public auto ansi beforefieldinit Invalid2<U,V> extends class GenTestType<!U,!V>
+{
+  .custom instance void [System.Private.CoreLib]System.Runtime.CompilerServices.ValidateMethodImplRemainsInEffectAttribute::.ctor() = (01 00 00 00)
```

Comment:
This should be added to `System.Runtime` ref assembly and referenced via `[System.Runtime]`. We should avoid adding tests that depend on `System.Private.CoreLib` directly. They cause a lot of complications.

Reply:
How can I add it to the `System.Runtime` facade?

Reply:
Add it to https://github.com/dotnet/runtime/tree/master/src/libraries/System.Runtime/ref